#!/usr/bin/env python
PACKAGE = "iris_cobot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("viz_cloud",            bool_t,     0, "Toogle between viz and cec cloud",  True)
gen.add("robot_models",         bool_t,     0, "Robot sphere models",               False)
gen.add("obstacle_models",      bool_t,     0, "Obstacle sphere models",            True)
gen.add("random_obstacles",     bool_t,     0, "Look for random obstacles",         True)
gen.add("sphere_obstacles",     bool_t,     0, "Look for obstacles sphere like",    False)

gen.add("min_dist_robot",       double_t,   0, "Minimum distance of robot",         0.1, 0.05, 1)
gen.add("min_dist_interest",    double_t,   0, "Minimum distance of interst",       0.3, 0.1,  1)

gen.add("leaf_size",            double_t,   0, "Voxel down sample leaf size",       0.03, 0.005, 1)
gen.add("radius_search",        double_t,   0, "Search readius for KDTree",         0.03, 0.005, 1)
gen.add("cluster_tolerance",    double_t,   0, "Clustering algorithm tolerance",    0.03, 0.005, 20)

gen.add("min_cluster_size",     int_t,      0, "Min cluster size (Points / X)",     10,  1, 100)
gen.add("max_cluster_size",     int_t,      0, "Max cluster size (Points / X)",     90,  1, 100)
gen.add("squared_dist",         double_t,   0, "Minimum squared distance",          0.001, 0.00000001, 10)
gen.add("normal_diff",          double_t,   0, "Minimum normal difference",         0.99, 0.001, 1)


exit(gen.generate(PACKAGE, "iris_cobot", "ObstacleDetection"))  